src/settings/Settings.cpp

```diff
@@ -0,0 +1,585 @@
+//Copyright (c) 2018 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#include <cctype>
+#include <fstream>
+#include <stdio.h>
+#include <sstream> // ostringstream
+#include <regex> // regex parsing for temp flow graph
+#include <string> //Parsing strings (stod, stoul).
+
+#include "Settings.h"
+#include "types/AngleDegrees.h" //For angle settings.
+#include "types/AngleRadians.h" //For angle settings.
+#include "types/Duration.h" //For duration and time settings.
+#include "types/LayerIndex.h" //For layer index settings.
+#include "types/Ratio.h" //For ratio settings and percentages.
+#include "types/Temperature.h" //For temperature settings.
+#include "types/Velocity.h" //For velocity settings.
+#include "../Application.h" //To get the extruders.
+#include "../utils/logoutput.h"
+
+namespace cura
+{
+
+void Settings::add(const std::string& key, const std::string value, ExtruderTrain* limit_to_extruder)
```

Comment:
I cannot find any place in the code where this function is called with a `limit_to_extruder`

Reply:
This was removed here: 71d03112cb8709ae7be4d89042555f7af53f9fa4